lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -1628,31 +1643,50 @@ struct ObjectSubfieldOpConversion
   const DenseMap<StringAttr, firrtl::ClassType> &classTypeTable;
 };
 
-struct ClassFieldOpConversion : public OpConversionPattern<ClassFieldOp> {
+LogicalResult convertClassFieldsTypes(ClassFieldsLike oldOp,
+                                      ClassFieldsLike newOp,
+                                      const TypeConverter *typeConverter) {
+  llvm::SmallVector<NamedAttribute> fieldTypes;
+  llvm::SmallVector<StringAttr> fieldNames = oldOp.getFieldNames();
+  mlir::DictionaryAttr fieldIdxs =
+      mlir::cast<mlir::DictionaryAttr>(oldOp->getAttr("fieldIdxs"));
+  for (auto field : fieldNames) {
+    std::optional<Type> type = oldOp.getFieldType(field);
+    if (!type.has_value())
+      return failure();
+    fieldTypes.push_back(mlir::NamedAttribute(
+        cast<StringAttr>(field),
+        mlir::TypeAttr::get(typeConverter->convertType(type.value()))));
+  }
+  auto *ctx = newOp.getContext();
+  newOp->setAttr("fieldNames", mlir::ArrayAttr::get(ctx, {fieldNames.begin(),
```

Comment:
This is somewhat obscure, but MLIR has a somewhat nifty helper for generically replacing attributes/types in an operation: https://github.com/llvm/llvm-project/blob/1e70122cbc187c08de91a3fb42843efb1221e0e9/mlir/include/mlir/IR/AttrTypeSubElements.h#L122-L134

Here's an example in CIRCT: https://github.com/llvm/circt/blob/eef76ca7e302ccef144343926c4e653a0e43a558/lib/Dialect/OM/Transforms/FreezePaths.cpp#L92-L100

You could potentially build an AttrTypeReplacer in the ClassFieldsOpConversion, which delegates to the typeConverter to do the replacements, and then call it's `replace` method on the newOp in matchAndRewrite. This point may be moot if we end up following the suggestion to put the field names and types on the ClassLike op, but thought I'd bring it up since it's a neat tool.